pioneer/packages/joy-roles/src/tabs/WorkingGroup.controller.tsx

```diff
@@ -4,45 +4,40 @@ import { Controller } from '@polkadot/joy-utils/react/helpers';
 import { View } from '@polkadot/joy-utils/react/hocs';
 
 import { ITransport } from '../transport';
+import { AvailableGroups } from '../working_groups';
 
-import { ContentCurators,
-  WorkingGroupMembership,
-  StorageProviders } from './WorkingGroup';
+import { WorkingGroupMembership,
+  ContentCurators,
+  StorageProviders,
+  OperationsGroup } from './WorkingGroup';
 
 import styled from 'styled-components';
-import { normalizeError } from '@polkadot/joy-utils/functions/misc';
 
 type State = {
-  contentCurators?: WorkingGroupMembership;
+  curators?: WorkingGroupMembership;
   storageProviders?: WorkingGroupMembership;
+  operationsGroup?: WorkingGroupMembership;
 }
 
 export class WorkingGroupsController extends Controller<State, ITransport> {
-  constructor (transport: ITransport, initialState: State = {}) {
+  constructor (transport: ITransport) {
     super(transport, {});
   }
 
   refreshState () {
-    this.getCurationGroup();
-    this.getStorageGroup();
+    void this.getGroups();
   }
 
-  getCurationGroup () {
-    this.transport.curationGroup()
-      .then((value: WorkingGroupMembership) => {
-        this.setState({ contentCurators: value });
-        this.dispatch();
-      })
-      .catch((e) => this.onError(normalizeError(e)));
-  }
+  async getGroups () {
+    const newState: Partial<State> = {};
 
-  getStorageGroup () {
-    this.transport.storageGroup()
-      .then((value: WorkingGroupMembership) => {
-        this.setState({ storageProviders: value });
-        this.dispatch();
+    await Promise.all(
+      AvailableGroups.map(async (group) => {
+        newState[group] = await this.transport.groupOverview(group);
```

Comment:
I know I'm not invited to this party, so do disregard this if it's inappropriate! I happen to be the original author of this controller pattern for this project and I saw this PR by chance.

Should this not be a `Promise.all()` sort of setup? Otherwise this loop will fetch the group overviews in series. not parallel.

Reply:
Hello Paul, thanks for the comment, but I'm not sure what you mean. This is a `Promise.all` setup and the queries should be executed in parallel, since I'm using `await Promise.all` on an array of promises created from `AvailableGroups`.

Reply:
Maybe I've misunderstood what's going on here (it's been a while since I looked at this codebase) but isn't the mapping of `AvailableGroups` assigning the variable `newState` but not returning a Promise?

In other words, don't you want

```
const groups = await Promise.all( AvailableGroups.map(group => this.transport.groupOverview(group)));
// ... add each group to newState
```

? Otherwise, isn't `AvailableGroups.map` waiting for the promise to resolve on each iteration?

Reply:
I'm passing an async function to `map`. An async function returns a promise (by definition) and all the promises returned by `map` are awaited on with `Promise.all`, so they are all executed in parallel.

You can try it out:
```
const a = () => new Promise(
  (resolve, reject) => {
    setTimeout(() => resolve(true), 1000)
  }
)

async function b () {
  const start = Date.now()
  await Promise.all(
    [1,2,3].map(async () => {
      await a()
    })
  )
  console.log(Date.now() - start)
}
```

Reply:
I did not know about that feature! Thanks, and sorry to bother you.